include/csp/csp_endian.h

```diff
@@ -32,6 +32,21 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 extern "C" {
 #endif
 
+/**
+   Convert from host order to network order.
+*/
+int16_t csp_shton16(int16_t h16);
+
+/**
+   Convert from host order to network order.
+*/
+int32_t csp_shton32(int32_t h32);
+
+/**
+   Convert from host order to network order.
+*/
+int64_t csp_shton64(int64_t h64);
+
```

Comment:
Why do they have 's'? If you wanna distinguish between user-defined and csp original functions, I'd recommend calling shton from an hton function. Because the inconsistency would make using them harder and more prone to mistakes.

Reply:
shton "signed host to network" and sadly you can't mix these functions together or have one call the other because they deal in different data types... The only way it could work is if you cast to unsigned, reverse byte order and cast back. which would work except we change the endianness. I'm not sure if you can cast an unsigned to signed when it's in the wrong endian for the system

Reply:
Oh, I see. I didn't realize it was for signed and the unsigned ones already exist. You don't need to do that. Keep it this way.